core/src/main/java/hudson/model/ParametersDefinitionProperty.java

```diff
@@ -71,17 +71,11 @@
 
     @DataBoundConstructor
     public ParametersDefinitionProperty(@Nonnull List<ParameterDefinition> parameterDefinitions) {
-        if (parameterDefinitions == null) {
-            throw new NullPointerException("ParameterDefinitions is null when this is a not valid value");
-        }
-        this.parameterDefinitions = parameterDefinitions;
+        this.parameterDefinitions = parameterDefinitions != null ? parameterDefinitions : new ArrayList();
```

Comment:
Should this use the Java-7 typed collection syntax:

> new ArrayList<>()

instead of the earlier syntax:

> new ArrayList()?

Reply:
Are there side effects from a `parameterDefinitions` being assigned the value `null` when previously it was not allowed to have a null value?

Reply:
> being assigned the value `null`

It's not.